### ISSUE TITLE
ExecuteDelete does not respect table mapping when also mapped to a view

### ISSUE THREAD

**Issue by statler** [dotnet/efcore#34834]
Since EF5 it has been possible to map to both a view and a table:
https://learn.microsoft.com/en-au/ef/core/modeling/entity-types?tabs=data-annotations#view-mapping

To quote the docs:
_In this case the query mapping will be used for queries and the table mapping will be used for updates._

This works as expected with add, update, delete and select on entities that aren't executed directly. E.g. this works:

```C#
    _context.Approvals.RemoveRange(context.Set<Approval>().Where(x => x.ProjectId == ProjectId).ToList());
    context.SaveChanges();
```

It does NOT work with ExecuteDelete which tries to execute delete on the view instead of the table giving the following exception

```C#
  await _context.Set<Approval>().Where(x => x.ProjectId == ProjectId).ExecuteDelete();
```

Microsoft.Data.SqlException: View or function 'a' is not updatable because the modification affects multiple base tables - which is the error you get if you map just to the view instead of the table.

Is there a workaround until it is fixed?

Related to https://github.com/dotnet/efcore/issues/34694

EF Core version: 7
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: (e.g. .NET 7.0)
Operating system: win 64
IDE: e.g. Visual Studio 2022 17.4

**Comment by roji** [comment]
Duplicate of #34677

**Comment by roji** [comment]
This has already been fixed for EF 10, but unfortunately the fix is too complex to backport to 9.0 or 8.0. If we get lots of feedback (which we haven't so far), we can reconsider backporting.
